regions_www/m/pct_shiny_funs.R

```diff
@@ -165,10 +170,27 @@ data_filter <- function(scenario, type){
 }
 
 # Return red if the data is negative
-negative_red <- function(dataset, scenario, type, round_digits = 0){
-  ifelse(round(dataset[[data_filter(scenario, type)]], round_digits) < 0, "red", "inherit")
+negative_red <- function(data, scenario, type, round_digits = 0){
+  ifelse(round(data[[data_filter(scenario, type)]], round_digits) < 0, "red", "inherit")
 }
 
+# Identify small cells in schools layer
+school_smallcell <- function(expression, return_tf = F, d = F){
```

Comment:
Neat, looks good. Can be rewritten more concisely as:

```r
school_smallcell2 <- function(x, return_tf = F, d = F){
  if(!return_tf) {
    if(d) res = case_when(x %in% 1:5 ~ "1 to 5", TRUE ~ as.character(x)) else
      res = case_when(x %in% 1:2 ~ "1 to 2", TRUE ~ as.character(x))
  } else {
    if(d) res = case_when(x %in% 1:5 ~ TRUE, TRUE ~ FALSE) else
      res = case_when(x %in% 1:2 ~ TRUE, TRUE ~ FALSE)
  }
  res
}
```

Here's a [reprex](https://www.tidyverse.org/help/) testing both, your one is fine, was just looking if it could be made neater but yours is more explicit which is good:

``` r
library(dplyr)
#> 
#> Attaching package: 'dplyr'
#> The following objects are masked from 'package:stats':
#> 
#>     filter, lag
#> The following objects are masked from 'package:base':
#> 
#>     intersect, setdiff, setequal, union
x = 1:6

# original way
school_smallcell <- function(expression, return_tf = F, d = F) {
  if (d) {
    smallcell <- (expression > 0 & expression <= 5)  # Upper bound to suppress cells if 5 in schools [d], otherwise 2 [z, rnet]
    expression[smallcell] <- "1 to 5"
  } else {
    smallcell <- (expression > 0 & expression <= 2)
    expression[smallcell] <- "1 or 2"
  }
  if (return_tf) {
    smallcell
  } else {
    expression
  }
}

# recode way
school_smallcell2 <- function(x, return_tf = F, d = F) {
  if (!return_tf) {
    if (d) 
      res = case_when(x %in% 1:5 ~ "1 to 5", TRUE ~ as.character(x)) else res = case_when(x %in% 1:2 ~ "1 to 2", TRUE ~ as.character(x))
  } else {
    if (d) 
      res = case_when(x %in% 1:5 ~ TRUE, TRUE ~ FALSE) else res = case_when(x %in% 1:2 ~ TRUE, TRUE ~ FALSE)
  }
  res
}
school_smallcell(x)
#> [1] "1 or 2" "1 or 2" "3"      "4"      "5"      "6"
school_smallcell(x, d = TRUE)
#> [1] "1 to 5" "1 to 5" "1 to 5" "1 to 5" "1 to 5" "6"
school_smallcell(x, return_tf = TRUE)
#> [1]  TRUE  TRUE FALSE FALSE FALSE FALSE
school_smallcell2(x)
#> [1] "1 to 2" "1 to 2" "3"      "4"      "5"      "6"
school_smallcell2(x, d = TRUE)
#> [1] "1 to 5" "1 to 5" "1 to 5" "1 to 5" "1 to 5" "6"
school_smallcell2(x, return_tf = TRUE)
#> [1]  TRUE  TRUE FALSE FALSE FALSE FALSE
```

Reply:
oh nice... happy for you to change it if you would like, I wasn't sure if on balance you thought conciseness or explicitness was better

Reply:
Aha, yes I assumed we were using the tidyverse as I use it everyday but agree better to keep it explicit in this case. Thanks for trying and sorry if its been a waste of time (hopefully an interesting waste of time hehe).